pkg/tsdb/elasticsearch/elasticsearch.go

```diff
@@ -2,50 +2,173 @@ package elasticsearch
+		im:                 im,
+		HTTPClientProvider: httpClientProvider,
+		intervalCalculator: tsdb.NewCalculator(),
 	}
 }
 
-// Query handles an elasticsearch datasource request
-//nolint: staticcheck // plugins.DataResponse deprecated
-func (e *Executor) DataQuery(ctx context.Context, dsInfo *models.DataSource,
-	tsdbQuery plugins.DataQuery) (plugins.DataResponse, error) {
-	if len(tsdbQuery.Queries) == 0 {
-		return plugins.DataResponse{}, fmt.Errorf("query contains no queries")
+func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	if len(req.Queries) == 0 {
+		return &backend.QueryDataResponse{}, fmt.Errorf("query contains no queries")
 	}
 
-	client, err := es.NewClient(ctx, e.httpClientProvider, dsInfo, *tsdbQuery.TimeRange)
+	dsInfo, err := s.getDSInfo(req.PluginContext)
 	if err != nil {
-		return plugins.DataResponse{}, err
+		return &backend.QueryDataResponse{}, err
 	}
 
-	if tsdbQuery.Debug {
-		client.EnableDebug()
+	client, err := es.NewClient(ctx, s.HTTPClientProvider, dsInfo, req.Queries[0].TimeRange)
+	if err != nil {
+		return &backend.QueryDataResponse{}, err
 	}
 
-	query := newTimeSeriesQuery(client, tsdbQuery, e.intervalCalculator)
+	query := newTimeSeriesQuery(client, req.Queries, s.intervalCalculator)
 	return query.execute()
 }
+
+func newInstanceSettings() datasource.InstanceFactoryFunc {
+	return func(settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+		jsonData := map[string]interface{}{}
+		err := json.Unmarshal(settings.JSONData, &jsonData)
+		if err != nil {
+			return nil, fmt.Errorf("error reading settings: %w", err)
+		}
+		httpCliOpts, err := settings.HTTPClientOptions()
+		if err != nil {
+			return nil, fmt.Errorf("error getting http options: %w", err)
+		}
+
+		version, err := coerceVersion(jsonData["esVersion"])
+
+		if err != nil {
+			return nil, fmt.Errorf("elasticsearch version is required, err=%v", err)
+		}
+
+		timeField, ok := jsonData["timeField"].(string)
+		if !ok {
+			return nil, errors.New("timeField cannot be cast to string")
+		}
+
+		if timeField == "" {
+			return nil, errors.New("elasticsearch time field name is required")
+		}
+
+		interval, ok := jsonData["interval"].(string)
+		if !ok {
+			interval = ""
+		}
+
+		timeInterval, ok := jsonData["timeInterval"].(string)
+		if !ok {
+			timeInterval = ""
+		}
+
+		maxConcurrentShardRequests, ok := jsonData["maxConcurrentShardRequests"].(float64)
+		if !ok {
+			maxConcurrentShardRequests = 256
+		}
+
+		model := es.DatasourceInfo{
+			ID:                         settings.ID,
+			URL:                        settings.URL,
+			HTTPClientOpts:             httpCliOpts,
+			Database:                   settings.Database,
+			MaxConcurrentShardRequests: int64(maxConcurrentShardRequests),
+			ESVersion:                  version,
+			TimeField:                  timeField,
+			Interval:                   interval,
+			TimeInterval:               timeInterval,
```

Comment:
What's the difference between these? Think only `timeInterval` is stored in jsonData.

Reply:
Seen both, that's why I kept them!

Reply:
Hmm okay. Sure you haven't seen `timeInterval` on the data source config and `interval` on each query?

Reply:
Yeah, I could see both on `JSONData`.

Have a look here for `interval`: https://github.com/grafana/grafana/pull/36132/files#diff-95c1bcdd942efc587e6f08b4175e52e30410f94425bc990a9260fb446f131831L90

Reply:
Oops you're absolutely correct. Forgot about that one. Unfortunate it's being named `interval` as well :joy: Great